sql-to-dbsp-compiler/lib/sqllib/src/string.rs

```diff
@@ -318,3 +320,217 @@ pub fn writelog<T: std::fmt::Display>(format: String, argument: T) -> T {
     print!("{}", formatted);
     argument
 }
+
+pub fn md5__(value: String) -> String{
+    let mut  hasher = Md5::new();
+    hasher.input(value.as_bytes());
+    let result = hasher.result();
+    return format!("{:x}", result);
+}
+
+pub fn to_lower_hex(value: i64) -> String{
+    format!("{:x}", value)
+}
+
+pub fn to_upper_hex(value: i64) -> String{
+    format!("{:X}", value)
+}
+
+pub fn translate_(value: String, from: String, to: String) -> String{
+    let mut  index =   0;
+    let len_of_to = to.len();
+    let mut tmp_value =  (&value).to_string();
+    for x in from.chars() {
+        if index < len_of_to{
+            let inn = index.clone();
+            let char_replace = to.chars().nth(inn).unwrap().to_string();
```

Comment:
doesn't seem obvious to me why it needs to call to_string here every time? Can't it do in-place mutation since you have ownership of all the strings?